### PR TITLE
Record-track: Trajectory-State Readout + Muon 0.98 + Legal TTT (1.0788)

### PR DESCRIPTION
**Pull request by aazizyan**
## Results

3-seed mean: **1.0788 BPB** (std 0.0003). All seeds under 16,000,000 bytes.

| Seed | Steps | Sliding BPB | TTT BPB | Artifact bytes |
|------|-------|-------------|---------|----------------|
| 42   | 4,623 | 1.0804      | 1.0787  | 15,990,973     |
| 2025 | 4,659 | 1.0804      | 1.0791  | 15,989,784     |
| 1234 | 4,657 | 1.0799      | 1.0785  | 15,990,480     |
| Mean | 4,646 | 1.0802      | 1.0788  |                |
| Std | | 0.0003 | 0.0003 |                 |

Against the current merged SOTA (PR #1493, @bigbag — 1.0810 BPB): **-0.0022 BPB**. The corresponding `val_loss` improvement is approximately `-0.006 nats`, which is above the leaderboard's stated `0.005-nat` magnitude threshold.

## What changed

Built on PR #1493. Full recipe differences vs PR #1493 are listed in the README.

The novel contribution is **trajectory-state readout** — tapping block-5 hidden states at all three recurrence passes and learning a grouped correction (16 groups × 2 scalars = 32 learned parameters) that recovers information from earlier trajectory states that is not fully preserved in the final pass. Identity at init, negligible overhead.

Also adopted from PR #1518 (@abaybektursun): MUON_MOMENTUM=0.98, GPTQ_RESERVE_SECONDS=4, GPTQ_CALIBRATION_BATCHES=16. Added step-based loop activation at step 2000 to fix a wallclock-based curriculum confound. QK_GAIN_INIT is 5.0 (code default, vs PR #1493's 5.25 — unintentional, not isolated).

## Negative results

Tested and rejected on this stack: Birkhoff contraction (appeared to reduce useful state diversity), FiLM gamma (absorbed by shared weights), per-pass embeddings (zero effect), controls-only TTT (regressed by +0.002 BPB), parallel_residual_start=8 (neutral). Details in README.

## Reproduction

```bash
TTT_ENABLED=1 MUON_MOMENTUM=0.98 \
    GPTQ_RESERVE_SECONDS=4 GPTQ_CALIBRATION_BATCHES=16 \
    ENABLE_LOOPING_AT_STEP=2000 \
    USE_PASS_READOUT=1 READOUT_GROUPS=16 READOUT_SCALE=0.5 \
    SEED=42 RUN_ID=train_seed42 \
    torchrun --standalone --nproc_per_node=8 train_gpt.py
```

## Compliance (Issue #1017, Track B)

Causal sliding-window eval, standard softmax, score-before-update TTT, single pass. No SLOT, no pre-quant TTT, no ETLB, no n-gram cache. Full details in README.

### CODE REVIEW

**Comment by aazizyan** [comment]
Some untested directions that might be worth exploring:

* Leaky trajectory readout (global scalar or one scalar per readout group) — let the model tune how aggressively it mixes early-pass information back into the final state
* Readout group sweep (`G=8` vs `16` vs `32`) — the current `G=16` choice was reasonable but not optimized, and the parameter cost stays tiny
* Quadratic grouped readout — add grouped second-order terms on top of the linear `p0/p1/p2` correction to see whether the trajectory signal is mildly nonlinear
* Readout-only TTT — controls-only TTT regressed on this stack, but adapting only `readout_delta` at eval time may still be worth a look
* Post-readout stabilization gate — one extra learned scalar on the correction term, mainly to test whether the current grouped readout is slightly too aggressive on some seeds
